projects/packages/forms/src/blocks/field-file/edit.js

```diff
@@ -0,0 +1,99 @@
+import { getJetpackExtensionAvailability } from '@automattic/jetpack-shared-extension-utils';
+import {
+	useBlockProps,
+	useInnerBlocksProps,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+import { useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import clsx from 'clsx';
+import JetpackFieldControls from '../shared/components/jetpack-field-controls';
+import { UpsellNudge } from '../shared/components/upsell-nudge';
+import useFormWrapper from '../shared/hooks/use-form-wrapper';
+import useJetpackFieldStyles from '../shared/hooks/use-jetpack-field-styles';
+import useParentFormClientId from '../shared/hooks/use-parent-form-client-id';
+import './editor.scss';
+
+export default function FileFieldEdit( props ) {
+	const { attributes, clientId, isSelected, setAttributes, name, className } = props;
+	const { id, required, width } = attributes;
+
+	const fieldFileAvailability = getJetpackExtensionAvailability( 'field-file' );
+
+	useFormWrapper( { attributes, clientId, name } );
+	const { blockStyle } = useJetpackFieldStyles( attributes );
+
+	const classes = clsx( className, 'jetpack-field is-non-animated-label', {
+		'is-selected': isSelected,
+		[ `jetpack-field__width-${ width }` ]: width,
+	} );
+
+	const blockProps = useBlockProps( {
+		className: classes,
+		style: blockStyle,
+	} );
+
+	const uploadLabel = __( 'Upload a file', 'jetpack-forms' );
+	const innerBlocksProps = useInnerBlocksProps( blockProps, {
+		template: [
+			[
+				'jetpack/label',
+				{
+					label: uploadLabel,
+					defaultLabel: uploadLabel,
+					lock: { move: true, remove: true },
+				},
+			],
+			[
+				'jetpack/dropzone',
+				{
+					lock: { move: true, remove: true },
+					layout: { type: 'flex', justifyContent: 'center', orientation: 'vertical' },
+				},
+			],
+		],
+		allowedBlocks: [ 'jetpack/dropzone', 'jetpack/label' ],
+		renderAppender: false,
+	} );
+
+	const formClientId = useParentFormClientId( clientId );
+
+	const selectedBlockClientId = useSelect( select => {
+		const { getSelectedBlockClientId } = select( blockEditorStore );
+		return getSelectedBlockClientId();
+	} );
+
+	const selectedFormClientId = useParentFormClientId( selectedBlockClientId );
+
+	const requiresCustomUpgradeNudge = useMemo( () => {
+		return (
+			( ! fieldFileAvailability || ! fieldFileAvailability.available ) &&
+			fieldFileAvailability?.unavailableReason?.includes( 'nudge_disabled' )
+		);
+	}, [ fieldFileAvailability ] );
+
+	const childern = innerBlocksProps.children;
+	delete innerBlocksProps.children;
```

Comment:
I think it would be best to avoid the mutating code. It's possible `innerBlocksProps` is being memoized/cached across renders, in which case deleting `children` in a mutating way might cause issues on later renders. This is an alternative:
```jsx
const { children, ...restInnerBlocksProps } = innerBlocksProps;

<div { ...restInnerBlocksProps }>
```

If you're ok with a wrapper element around `children` I think you can also do the following, which seems more in spirit of the `useInnerBlocksProps` API:
- Not pass `blockProps` to `useInnerBlocksProps`
- Spread `blockProps on the wrapper `div` in place of `innerBlocksProps`
- Add a wrapper around `{ children }` - `<div { ...innerBlocksProps }>{ children }</div>`

Reply:
Here's some pseudocode:
```jsx
const blockProps = useBlockProps( // ... );
const innerBlockProps = useInnerBlockProps( // ... );

return (
    <div { ...blockProps }>
        { requiresCustomUpgradeNudge && ( // ... ) }
        <div { ...innerBlockProps }>{children}</div>
    </div>
);
```